src/main/java/com/solidbrain/WorkflowTask.java

```
@@ -0,0 +1,284 @@
+                createNewDeal(contact);
+            }
+        }
+        return true;
+    }
+
+    private boolean processDeal(Meta meta, Deal deal) {
+        log.trace("Current deal={}", deal);
+
+        String eventType = meta.getSync().getEventType();
+        log.trace("eventType={}", eventType);
+        if (eventType.contentEquals(createdEventName) || eventType.contentEquals(updatedEventName)) {
+            log.debug("Deal sync eventType={}", eventType);
+
+            verifyExistingDeal(deal);
+        }
+        return true;
+    }
+
+    private void verifyExistingDeal(Deal deal) {
+        if (isDealStageWon(deal)) {
+            log.info("Verifying deal in Won stage");
+            log.debug("Deal={}", deal);
+
+            Contact dealsContact = fetchExistingContact(deal.getContactId());
+            log.trace("Deal's contact={}", dealsContact);
+
+            User contactOwner = fetchOwner(dealsContact.getOwnerId());
+            log.trace("Contact's owner={}", contactOwner);
+
+            if (!isContactOwnerAnAccountManager(contactOwner)) {
+                updateExistingContact(dealsContact);
+            }
+        }
+    }
+
+    private boolean updateExistingContact(Contact dealsContact) {
+        log.info("Updating contact's owner");
+
+        Long accountManagerId = Optional.ofNullable(getUserIdByName(favouriteAccountManagerName))
+                .orElseThrow(() -> new NoSuchElementException("User " + favouriteAccountManagerName + " not available"));
+
+        log.trace("accountManagerId={}", accountManagerId);
+
+        Map<String, Object> contactAttributes = new HashMap<>();
+        contactAttributes.put("owner_id", accountManagerId);
+        Contact updatedContact = baseClient.contacts()
+                                        .update(dealsContact.getId(), contactAttributes);
+
+        log.debug("Updated contact={}", updatedContact);
+
+        return true;
+    }
+
+    private Long getUserIdByName(String name) {
```

First of all we do not need this method at all... but if it is already here...

there are 2 options:
 * if we expect user to always be there then it should throw exception if user is not found
 * if we accept that user is there or is not there then maybe it should return `Optional<Long>`?

Another thing is that it probably should return `User` not its id.

---

Fixed using Optional and throwing MissingResourceException (see one of previous comments)